packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -114,6 +102,22 @@ const experimentalStyled = (tag, options, muiOptions = {}) => {
       });
     }
 
+    if (Array.isArray(styleArg)) {
+      // If the type is array, than we need to add placeholders in the template for the overrides, variants and the sx styles
+      transformedStyleArg = [
+        ...styleArg,
+        ...new Array(expressionsWithDefaultTheme.length).fill(''),
```

Comment:
```suggestion
        ...new Array(expressionsWithDefaultTheme.length - expressions.length).fill(''),
```

I'd say this is the correct calculation based on the previous logic. We should only add placeholders for the things we are manually adding, in this case I think this is the correct calculation.

Let me verify tomorrow, just to make sure.

Reply:
Yep, this is the correct behavior. For the expressions we should not alter anything, Please apply the suggestion on both places.